org.eclipse.xtext.builder.tests/src/org/eclipse/xtext/builder/tests/SharedInjectorProvider.java

```diff
@@ -0,0 +1,24 @@
+/*******************************************************************************
+ * Copyright (c) 2019 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.builder.tests;
+
+import org.eclipse.xtext.testing.IInjectorProvider;
+
+import com.google.inject.Injector;
+
+/**
+ * @author Sebastian Zarnekow - Initial contribution and API
+ */
+public class SharedInjectorProvider implements IInjectorProvider {
```

Comment:
can this be a singleton?

Reply:
It's not injected but used with `@InjectWith` similar to the generated injector providers.

Reply:
hmmm looks like i was not completely awake